applicationinsights/channel/AsynchronousSender.py

```diff
@@ -123,6 +129,14 @@ def _run(self):
                 # if we didn't get any items from the queue, we're done here
                 if len(data) == 0:
                     break
+                
+                # Drop extra events from the queue until it no larger than send_buffer_size.
+                # This is a necessary evil to prevent the backlog from bloating forever in bandwidth-limited environments
+                eventsToDrop = max(0, local_queue._queue.qsize() - self._send_buffer_size)
```

Comment:
I'm fine with setting "safe" high number like 50000. I just don't want it to be infinite by default so that developers get bitten in the butt for not knowing the internal details of this code